### ISSUE TITLE
Add report of http streaming in mixer filter

### ISSUE THREAD

**Issue by bianpengyuan** [istio/proxy#1828]
Right now there is http streaming is treated the same as unary http, which means there is only one report at stream closing. With envoy v2 starting to use grpc streaming, it's imperative to add telemetry report for http streaming. Suggest we start with something like tcp, which periodically reports inbound outbound bytes.

cc @qiwzhang

**Comment by kyessenov** [comment]
Do we collect TCP measurements (bytes) at the moment for HTTP? If so,
perhaps we should just share the bits that send them on timer between the
two mixer filters.

**Comment by qiwzhang** [comment]
I don't think we add tcp mixer filter for HTTP listeners.   Even so,  it is still different.  A tcp mixer-filter only for the tcp connection.  But a grpc-stream is different with a tcp connection.   A tcp connection can have multiple long-live grpc streams.

**Comment by douglas-reid** [comment]
We explicitly chose not to enable TCP filters and HTTP filters at the same time. It helped eliminate having to avoid multiple Check() calls (and other issues).

I feel like for the streaming case, we are going to need a bit of design work involving some new attributes (and values for things like `context.protocol`).
